### DISCUSSION TITLE
Updating the Slovak translation

### DISCUSSION THREAD

**Discussion by Svec-Tomas** [sparkle-project/Sparkle#2654] [Ideas]
Hey Sparkle Team!

I wonder if there is somewhere hidden a guide or how-to start if you want to translate / update an already existing translation. The Slovak translation seems to be incomplete and I don't know which file I should start to work on.

In the [en directory](https://github.com/sparkle-project/Sparkle/tree/2.x/Sparkle/en.lproj) there is no Sparkle.strings file for translation like for other languages. In the [sk folder](https://github.com/sparkle-project/Sparkle/blob/2.x/Sparkle/sk.lproj/Sparkle.strings) the Sparkle.strings file has only 80 lines, while the [de folder](https://github.com/sparkle-project/Sparkle/blob/2.x/Sparkle/de.lproj/Sparkle.strings) 200 lines of code.

Can someone let me know please the most up to date file(s) for translation?

Thanks a lot!

**Comment by zorgiepoo** [comment]
The remaining English localizations (Sparkle.strings) are in `Base.lproj` (I should re-organize this so the `SUUpdateAlert` and `SUUpdatePermissionPrompt` strings aren't in the en directory..)

We also have a script that checks the localizations. I have rarely run this program myself, but for the missing Sparkle.strings for Slovak it yields:

```
Missing %1$@ %2$@ has been downloaded and is ready to use! This is an important update; would you like to install it and relaunch %1$@ now?
Missing %1$@ %2$@ is available but your macOS version is too new for this update. This update only supports up to macOS %3$@.
Missing %1$@ %2$@ is available but your macOS version is too old to install it. At least macOS %3$@ is required.
Missing %1$@ can’t be updated if it’s running from the location it was downloaded to.
Missing %@ %@ is now available—you have %@. This is an important update; would you like to download it now?
Missing %@ %@ is now available—you have %@. Would you like to learn more about this update on the web?
Missing %@ is now updated to version %@!
Missing %@ is now updated!
Missing An error occurred while connecting to the installer. Please try again later.
Missing An error occurred while launching the installer. Please try again later.
Missing An error occurred while running the updater. Please try again later.
Missing An error occurred while starting the installer. Please try again later.
Missing An important update to %@ is ready to install
Missing Application Name
Missing Application Version
Missing CPU Speed (MHz)
Missing CPU Subtype
Missing CPU Type
Missing CPU is 64-Bit?
Missing Failed to resume installing update.
Missing Install on Quit
Missing Interactive based packages cannot be installed as the root user.
Missing Learn More…
Missing Mac Model
Missing Memory (MB)
Missing No
Missing Number of CPUs
Missing OS Version
Missing Preferred Language
Missing Quit %1$@, move it into your Applications folder, relaunch it from there and try again.
Missing The installation failed due to not having permission to write the new update.
Missing The update checker failed to start correctly. You should contact the app developer to report this issue and verify that you have the latest version.
Missing The update is improperly signed and could not be validated. Please try again later or contact the app developer.
Missing Unable to Check For Updates
Missing Update Installed
Missing Version History
Missing Yes
Missing You may need to allow modifications from %1$@ in System Settings under Privacy & Security and App Management to install future updates.
Missing Your macOS version is too new
Missing Your macOS version is too old
Unlocalized OK
```

Some of these are new to 2.7 (for the anonymous system profiler information) which is not released yet.

Check the SUUpdateAlert and SUUpdatePermissionPrompt separately.

**Reply by Svec-Tomas** [reply]
Could you please share the most up-to-date files? I'd like to fine-tune some of the existing translations as well. It would be great to have all three files with all the strings for translation.

**Reply by zorgiepoo** [reply]
English files which should be up to date:
`Sparkle/Sparkle/en.lproj/SUUpdateAlert.strings`
`Sparkle/Sparkle/en.lproj/SUUpdatePermissionPrompt.strings`
`Sparkle/Sparkle/Base.lproj/Sparkle.strings`

Slovak files:
`Sparkle/Sparkle/sk.lproj/Sparkle.strings` (needs updating, see above post for which localizations are missing)
`Sparkle/Sparkle/sk.lproj/SUUpdateAlert.strings` (already up to date but you can verify existing localizations if you want)
`Sparkle/Sparkle/sk.lproj/SUUpdatePermissionPrompt.strings` (missing `Automatically download and install updates`)

Clone the repo with git.
Create a pull request (e.g. look at #2504)
